app/utils/guaranteeParsing/GuaranteeXmlReaders.scala

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package utils.guaranteeParsing
+
+import cats.data.ReaderT
+import models.ParseError._
+import models._
+
+import scala.util.{Failure, Success, Try}
+import scala.xml.{Node, NodeSeq}
+
+class GuaranteeXmlReaders extends ParseHandling {
+
+  def parseGuarantees(xml: NodeSeq): ParseHandler[Seq[Guarantee]] = {
+
+    val guaranteeEithers: Seq[Either[ParseError, Guarantee]] =
+      (xml \ "GUAGUA").map {
+        node => guarantee(node) match {
+          case Left(e) => Left(e)
+          case Right(g) => Right(g)
+        }
+      }
+
+    ParseError.liftParseError(guaranteeEithers)
+  }
+
+  def parseSpecialMentions(xml: NodeSeq): ParseHandler[Seq[SpecialMention]] = {
+    ParseError.liftParseError((xml \ "SPEMENMT2").map {
+      node =>
+        specialMention(node) match {
+          case Left(e) => Left(e)
+          case Right(s) => Right(s)
+        }
+    })
+  }
+
+  val gooBlock: ReaderT[ParseHandler, NodeSeq, Seq[GooBlock]] =
```

Comment:
I'm not sure if there are going to be multiple GooBlock, shouldn't there be just one?

Reply:
It looks like we're being forced to call head in `EnsureGuaranteeService` because we have a Seq[GooBlock] here

Reply:
We can have multiple GooBlocks in a single xml document. At least in the examples I've been given.